cli/src/lib/token-storage.ts

```diff
@@ -0,0 +1,198 @@
+import envPaths from "env-paths";
+import {
+  chmodSync,
+  existsSync,
+  mkdirSync,
+  readFileSync,
+  unlinkSync,
+  writeFileSync,
+} from "fs";
+import { join } from "path";
+
+/**
+ * Token data structure stored on disk
+ */
+export interface StoredToken {
+  /** The session token for API authentication */
+  sessionToken: string;
+  /** When the token expires (ISO string) */
+  expiresAt: string;
+  /** User information */
+  user: {
+    id: string;
+    email: string | null;
+    name: string | null;
+  };
+  /** When the token was stored (ISO string) */
+  storedAt: string;
+}
+
+/**
+ * Get OS-specific paths for tambo data storage
+ * Uses XDG Base Directory Specification on Linux
+ *
+ * Data paths (for auth tokens and persistent data):
+ * - macOS: ~/Library/Application Support/tambo/
+ * - Linux: ~/.local/share/tambo/ (or XDG_DATA_HOME)
+ * - Windows: %LOCALAPPDATA%/tambo/
+ */
+const paths = envPaths("tambo", { suffix: "" });
+
+/**
+ * Path to the auth token file
+ */
+const AUTH_FILE_NAME = "auth.json";
+
+function getAuthFilePath(): string {
+  return join(paths.data, AUTH_FILE_NAME);
+}
+
+/**
+ * Ensure the data directory exists with appropriate permissions
+ */
+function ensureDataDir(): void {
+  const dataDir = paths.data;
+  if (!existsSync(dataDir)) {
+    mkdirSync(dataDir, { recursive: true, mode: 0o700 });
+  }
+}
```

Comment:
I'm going to be a bit pedantic here and say I want to support XDG_ spec on macos too.


```suggestion
/**
 * Name of directory to store state/config/cache
 */
const DIR_PREFIX = 'tambo';

/**
 * Get OS-specific paths for tambo state storage
 * Uses XDG Base Directory Specification on Linux
 *
 * Data paths (for auth tokens and persistent data):
 * - macOS: ~/Library/Application Support/tambo/ (or $XDG_DATA_HOME/tambo/)
 * - Linux: ~/.local/share/tambo/ (or $XDG_DATA_HOME/tambo/)
 * - Windows: %LOCALAPPDATA%/tambo/
 */
function getDir(type: 'cache' | 'config' | 'data'): string {
  const paths = envPaths(DIR_PREFIX, { suffix: "" });
  const dir = process.env[`TAMBO_${type.toUpperCase()}_DIR`] || process.env[`XDG_${type.toUpperCase()}_HOME`] || paths[type];
  return join(dir, DIR_PREFIX);
}

/**
 * Path to the auth token file
 */
const AUTH_FILE_NAME = "auth.json";

function getAuthFilePath(): string {
  return join(getDir('data'), AUTH_FILE_NAME);
}

/**
 * Ensure the data directory exists with appropriate permissions
 */
function ensureDataDir(): void {
  const dataDir = getDir('data');
  if (!existsSync(dataDir)) {
    mkdirSync(dataDir, { recursive: true, mode: 0o700 });
  }
}
```

Reply:
made these changes too!